macros/utils/data_types/cast_column.sql

```diff
@@ -1,7 +1,19 @@
 {%- macro edr_cast_as_timestamp(timestamp_field) -%}
+    {{ return(adapter.dispatch('edr_cast_as_timestamp', 'elementary')(timestamp_field)) }}
+{%- endmacro -%}
+
+{%- macro default__edr_cast_as_timestamp(timestamp_field) -%}
     cast({{ timestamp_field }} as {{ elementary.edr_type_timestamp() }})
 {%- endmacro -%}
 
+{# Athena needs explicit conversion for ISO8601 timestamps used in buckets_cte #}
```

Comment:
`agate_to_dicts` converts timestamps to ISO8601 format and Athena needs explicit handling